engine/src/cmd/basecomputer.cpp

```diff
@@ -396,8 +397,25 @@ static float basicRepairPrice(void) {
     return price * g_game.difficulty;
 }
 
-static float SellPrice(float operational, float price) {
-    return usedValue(price);// - RepairPrice(operational, price);
+static double GetOperational(Unit *playerUnit, const Cargo *item) {
+    ComponentType type = GetComponentTypeFromName(item->GetName());
+    Component* component = playerUnit->GetComponentByType(type);
+
+    if(component) {
+        return component->PercentOperational();
+    }
+
+    return 0.0;
+}
+
+static double RepairPrice(Unit *playerUnit, const Cargo *item) {
+    // TODO: * configuration()->general.difficulty;
+    return 0.8 * (1-GetOperational(playerUnit, item)) * item->GetPrice();
```

Comment:
0.8 is a magic number. You might want to make it a named constant. Something like:

```suggestion
    constexpr double kRepairPriceCoefficient = 0.8;
    return kRepairPriceCoefficient * (1-GetOperational(playerUnit, item)) * item->GetPrice();
```

Reply:
It replaced 0.5 as the magic number and I incorrectly assumed the TODO was about that. 
This was a rush job. 
I'll fix this tomorrow before I test it.